src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Integer.cs

```diff
@@ -0,0 +1,166 @@
+            ReadInitialByte(out CborDataItem header);
+
+            switch (header.MajorType)
+            {
+                case CborMajorType.UnsignedInteger:
+                    value = ReadUnsignedInteger(in header, out int additionalBytes);
+                    AdvanceBuffer(1 + additionalBytes);
+                    return true;
+
+                default:
+                    value = 0;
+                    return false;
+            }
+        }
+
+        // Implements major type 0,1 decoding per https://tools.ietf.org/html/rfc7049#section-2.1
+        public long ReadInt64()
+        {
+            long value;
+            int additionalBytes;
+
+            ReadInitialByte(out CborDataItem header);
+
+            switch (header.MajorType)
+            {
+                case CborMajorType.UnsignedInteger:
+                    value = checked((long)ReadUnsignedInteger(in header, out additionalBytes));
+                    AdvanceBuffer(1 + additionalBytes);
+                    return value;
+
+                case CborMajorType.NegativeInteger:
+                    value = checked(-1 - (long)ReadUnsignedInteger(in header, out additionalBytes));
+                    AdvanceBuffer(1 + additionalBytes);
+                    return value;
+
+                default:
+                    throw new ArgumentException("Data item type mismatch");
+            }
+        }
+
+        public bool TryReadInt64(out long value)
+        {
+            ulong result;
+            int additionalBytes;
+
+            ReadInitialByte(out CborDataItem header);
+
+            switch (header.MajorType)
+            {
+                case CborMajorType.UnsignedInteger:
+                    result = ReadUnsignedInteger(in header, out additionalBytes);
+                    if (result > long.MaxValue)
+                    {
+                        value = 0;
+                        return false;
+                    }
+                    else
+                    {
+                        value = (long)result;
+                        AdvanceBuffer(1 + additionalBytes);
+                        return true;
+                    }
+
+                case CborMajorType.NegativeInteger:
+                    result = ReadUnsignedInteger(in header, out additionalBytes);
+                    if (result > long.MaxValue)
+                    {
+                        value = 0;
+                        return false;
+                    }
+                    else
+                    {
+                        value = -1 - (long)result;
+                        AdvanceBuffer(1 + additionalBytes);
+                        return true;
+                    }
+
+                default:
+                    value = 0;
+                    return false;
+            }
+        }
+
+        // Unsigned integer decoding https://tools.ietf.org/html/rfc7049#section-2.1
+        private ulong ReadUnsignedInteger(in CborDataItem header, out int additionalBytes)
```

Comment:
`CborDataItem` is only one byte, passing it by-ref is wasteful, just pass it by value.

Reply:
I expect the type to grow, particularly when nestedness is included.

Reply:
Unless it's going to exceed ~24 bytes, `in` is usually more expensive than not.

If it's needed later, that can be determined via performance testing... but until then you should just pass by value.